src/components/slices/CommentsSlice.ts

```diff
@@ -0,0 +1,182 @@
+  deleteComment,
+} from '../../api/comments';
+
+type NewComment = {
+  postId: number;
+  name: string;
+  email: string;
+  body: string;
+};
+
+export type CommentPost = {
+  comments: Comment[];
+  newComment: NewComment;
+  send: boolean;
+  formVisibility: boolean;
+  loading: boolean;
+  loadingNewComment: boolean;
+  error: string | null;
+};
+
+const initialState: CommentPost = {
+  comments: [],
+  newComment: {
+    postId: 0,
+    name: '',
+    email: '',
+    body: '',
+  },
+  send: true,
+  formVisibility: false,
+  loading: false,
+  loadingNewComment: false,
+  error: null,
+};
+
+export const getComments = createAsyncThunk(
+  '/getComments',
+  (postId: number) => {
+    return getPostComments(postId);
+  },
+);
+
+export const addComment = createAsyncThunk(
+  '/addComment',
+  (comment: NewComment) => {
+    return createComment(comment);
+  },
+);
+
+export const removeComment = createAsyncThunk(
+  '/removeComment',
+  (commentId: number) => {
+    return deleteComment(commentId);
+  },
+);
+
+export const CommentsSlice = createSlice({
+  name: 'comments',
+  initialState,
+  reducers: {
+    setComments: (state, action: PayloadAction<Comment[]>) => {
+      const currentState = state;
+
+      currentState.comments = action.payload;
+    },
+    setFormVisibility: (state, action: PayloadAction<boolean>) => {
+      const currentState = state;
+
+      currentState.formVisibility = action.payload;
+    },
+    setName: (state, action: PayloadAction<string>) => {
+      const currentState = state;
+
+      currentState.newComment.name = action.payload;
+    },
+    setEmail: (state, action: PayloadAction<string>) => {
+      const currentState = state;
+
+      currentState.newComment.email = action.payload;
+    },
+    setBody: (state, action: PayloadAction<string>) => {
+      const currentState = state;
+
+      currentState.newComment.body = action.payload;
+    },
+    setPostId: (state, action: PayloadAction<number>) => {
+      const currentState = state;
+
+      currentState.newComment.postId = action.payload;
+    },
+    setSend: (state, action: PayloadAction<boolean>) => {
+      const currentState = state;
+
+      currentState.send = action.payload;
+    },
+    setClear: state => {
+      const currentState = state;
+
+      currentState.newComment.body = '';
+      currentState.newComment.name = '';
+      currentState.newComment.email = '';
+      currentState.send = true;
+    },
+  },
+  extraReducers: builder => {
+    builder.addCase(getComments.pending, state => {
+      const currentState = state;
+
+      currentState.loading = true;
+      currentState.formVisibility = false;
+      currentState.error = null;
+    });
+    builder.addCase(getComments.fulfilled, (state, action) => {
+      const currentState = state;
+
+      currentState.comments = action.payload;
+      currentState.loading = false;
+    });
+    builder.addCase(getComments.rejected, state => {
+      const currentState = state;
```

Comment:
you can just use state everywhere

Reply:
I tried, but I'm getting an ESLint error.
"Assignment to property of function parameter 'state'.eslint[no-param-reassign](https://eslint.org/docs/rules/no-param-reassign)"

Reply:
Ah, I found the solution. I'll make a new commit soon.